sky/serve/replica_managers.py

```diff
@@ -910,21 +937,16 @@ def _terminate_replica(self,
                                               info)
             launch_thread = self._launch_thread_pool[replica_id]
             if launch_thread.is_alive():
-                request_id_found = False
-                # Should found immediately but add a 1 second timeout
-                # to avoid blocking the thread.
-                for _ in range(10):
+                self._replica_to_launch_cancelled[replica_id] = True
+                while True:
```

Comment:
We should add a check to `launch_thread.is_alive()` in this loop to avoid the case where the thread exits immediately after we check above - in that case it will never set the request_id or unset the cancel signal.